src/main/java/com/metamx/metrics/cgroups/ProcCgroupDiscoverer.java

```diff
@@ -0,0 +1,237 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.metrics.cgroups;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Files;
+import com.metamx.common.RE;
+import com.metamx.common.StringUtils;
+import com.metamx.metrics.CgroupUtil;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ProcCgroupDiscoverer implements CgroupDiscoverer
+{
+  private static final String CGROUP_TYPE = "cgroup";
+  private static final String PROC_TYPE = "getProc";
+
+  @Override
+  public Path discover(final String cgroup, long pid)
+  {
+    Preconditions.checkNotNull(cgroup, "cgroup required");
+    // Wish List: find a way to cache these
+    final File proc = getProc();
+    final File procMounts = new File(proc, "mounts");
+    final File procCgroups = new File(proc, "cgroups");
+    final File pidCgroups = new File(new File(proc, Long.toString(pid)), "cgroup");
+    final ProcCgroupsEntry procCgroupsEntry = getCgroupEntry(procCgroups, cgroup);
+    final ProcMountsEntry procMountsEntry = getMountEntry(procMounts, cgroup);
+    final ProcPidCgroupEntry procPidCgroupEntry = getPidCgroupEntry(pidCgroups, procCgroupsEntry.hierarchy);
+    final File cgroupDir = new File(
+        procMountsEntry.path.toFile(),
+        procPidCgroupEntry.path
+    );
+    if (cgroupDir.exists() && cgroupDir.isDirectory()) {
+      return cgroupDir.toPath();
+    }
+    throw new RE("Invalid cgroup directory [%s]", cgroupDir);
+  }
+
+  @VisibleForTesting
+  public File getProc()
+  {
+    // Wish List: discover `/getProc` in a more reliable way
+    final File proc = Paths.get("/proc").toFile();
+    Path foundProc = null;
+    if (proc.exists() && proc.isDirectory()) {
+      // Sanity check
+      try {
+        for (final String line : Files.readLines(new File(proc, "mounts"), Charsets.UTF_8)) {
+          final ProcMountsEntry entry = ProcMountsEntry.parse(line);
+          if (PROC_TYPE.equals(entry.type)) {
+            if (proc.toPath().equals(entry.path)) {
+              return proc;
+            } else {
+              foundProc = entry.path;
+            }
+          }
+        }
+      }
+      catch (IOException e) {
+        // Unlikely
+        throw Throwables.propagate(e);
```

Comment:
@drcrallen yes, but eventually Druid will have to refuse from using it because `Throwables.propage()` is going to be removed in Guava 22: https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate